src/hayhooks/server/utils/yaml_utils.py

```diff
@@ -65,20 +98,23 @@ def _resolve_declared_inputs(
     """
     resolutions: dict[str, InputResolution] = {}
     for io_name, declared_path in declared_map.items():
-        normalized_path = _normalize_declared_path(declared_path)
-        if not isinstance(normalized_path, str) or "." not in normalized_path:
+        candidate_paths = _collect_candidate_paths(declared_path)
+        if not candidate_paths:
             continue
 
+        normalized_path = candidate_paths[0]
         component_name, field_name = normalized_path.split(".", 1)
         meta = (pipeline_meta.get(component_name, {}) or {}).get(field_name, {}) or {}
         resolved_type = meta.get("type")
+        is_required = True
```

Comment:
Could you please explain this point? This applies to all inputs? The previous logic using `is_mandatory` was incorrect?